pkg/migration/controllers/resourcetransformation.go

```diff
@@ -0,0 +1,320 @@
+		}
+		for _, path := range spec.Paths {
+			// TODO: this can be validated via CRDs as well, when we have defined schema
+			// for stork crds
+			if path.Operation == stork_api.JsonResourcePatch {
+				return fmt.Errorf("json patch for resources is not supported, operation: %s", path.Operation)
+			}
+			if !(path.Operation == stork_api.AddResourcePath || path.Operation == stork_api.DeleteResourcePath) {
+				return fmt.Errorf("unsupported resource patch operation given for kind :%s, operation: %s", kind, path.Operation)
+			}
+			if !(path.Type == stork_api.BoolResourceType || path.Type == stork_api.IntResourceType ||
+				path.Type == stork_api.StringResourceType || path.Type == stork_api.SliceResourceType ||
+				path.Type == stork_api.KeyPairResourceType) {
+				return fmt.Errorf("unsupported type for resource %s, path %s, type: %s", kind, path.Path, path.Type)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ResourceTransformationController) validateTransformResource(ctx context.Context, transform *stork_api.ResourceTransformation) error {
+	resourceCollectorOpts := resourcecollector.Options{}
+	for _, spec := range transform.Spec.Objects {
+		gvk := strings.Split(spec.Resource, "/")
+		if len(gvk) != 3 {
+			return fmt.Errorf("invalid resource type should be in format <group>/<version>/<kind>, actual: %s", gvk)
+		}
+		kind := gvk[2]
+		resource := metav1.APIResource{
+			Name:       strings.ToLower(inflect.Pluralize(kind)),
+			Kind:       kind,
+			Version:    gvk[1],
+			Namespaced: true,
+			Group:      gvk[0],
+		}
+		log.TransformLog(transform).Infof("querying resource: %v", resource)
+		objects, err := r.resourceCollector.GetResourcesForType(
+			resource,
+			nil,
+			[]string{transform.Namespace},
+			spec.Selectors,
+			nil,
+			false,
+			resourceCollectorOpts,
+		)
+		if err != nil {
+			r.recorder.Event(transform,
+				v1.EventTypeWarning,
+				string(stork_api.ResourceTransformationStatusFailed),
+				fmt.Sprintf("Error getting resource kind:%s, err: %v", kind, err))
+			log.TransformLog(transform).Errorf("Error getting resources kind:%s, err: %v", kind, err)
+			return err
+		}
+		for _, path := range spec.Paths {
+			if !(path.Operation == stork_api.AddResourcePath || path.Operation == stork_api.DeleteResourcePath) {
+				return fmt.Errorf("unsupported operation type fo given path : %s", path.Operation)
+			}
+			for _, object := range objects.Items {
+				content := object.UnstructuredContent()
+				metadata, err := meta.Accessor(object)
+				if err != nil {
+					log.TransformLog(transform).Errorf("Unable to read metadata for resource %v, err: %v", kind, err)
+					return err
+				}
+				resInfo := &stork_api.TransformResourceInfo{
+					Name:             metadata.GetName(),
+					Namespace:        metadata.GetNamespace(),
+					GroupVersionKind: metav1.GroupVersionKind(object.GetObjectKind().GroupVersionKind()),
+				}
+				// TODO: add support for merge
```

Comment:
nit: ticket # tracking this

Reply:
decided to add support for merge, handled in next PR which is based on this branch